apps/dashboard/src/app/knowledge-box/utils.ts

```diff
@@ -7,3 +9,36 @@ export const KB_ROLE_TITLES: { [role in KBRoles]: string } = {
   SCONTRIBUTOR: 'generic.contributor',
   SMEMBER: 'generic.member',
 };
+
+export function getDesktopPlatform(): 'win' | 'mac' | 'linux' | null {
+  const platform = window.navigator.platform.toLowerCase();
+  if (platform.includes('win')) {
+    return 'win';
+  } else if (platform.includes('mac')) {
+    return 'mac';
+  } else if (platform.includes('linux')) {
+    return 'linux';
+  }
+  return null;
+}
+
+export function getDesktopAppUrl(platform: 'mac' | 'win' | 'linux'): Observable<string | null> {
+  const filenames = {
+    mac: /.*-universal-mac\.zip$/,
+    win: /.*\.exe$/,
+    linux: /.*\.snap$/,
+  };
```

Comment:
usually it's better to define regexp once for all (it's better for performances), so you can move this const outside of this function (no need to export it though).

Reply:
fixed 👍